compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -473,42 +474,85 @@ ireeGPUTargetInfoGet(MlirContext mlirCtx, const char *arch,
+  for (size_t i = 0; i < numMmaIntrinsics; i++) {
+    mma_intrinsic_t enumValue = mmaIntrinsics[i];
 
-      std::optional<mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsic>
-          virtualMmaIntrinsic =
-              mlir::iree_compiler::IREE::GPU::symbolizeVirtualMMAIntrinsic(
-                  enumValue);
-      if (virtualMmaIntrinsic) {
-        auto virtualMmaIntrinsicAttr =
-            mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsicAttr::get(
-                context, *virtualMmaIntrinsic);
-        mmaIntrinsicAttrs.push_back(virtualMmaIntrinsicAttr);
-        continue;
-      }
+    std::optional<mlir::iree_compiler::IREE::GPU::MMAIntrinsic> mmaIntrinsic =
+        mlir::iree_compiler::IREE::GPU::symbolizeMMAIntrinsic(enumValue);
+    if (mmaIntrinsic) {
+      auto mmaIntrinsicAttr =
+          mlir::iree_compiler::IREE::GPU::MMAIntrinsicAttr::get(context,
+                                                                *mmaIntrinsic);
+      mmaIntrinsicAttrs.push_back(mmaIntrinsicAttr);
+      continue;
+    }
 
-      assert(false &&
-             ("Invalid MMA intrinsic value: " + std::to_string(enumValue))
-                 .c_str());
+    std::optional<mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsic>
+        virtualMmaIntrinsic =
+            mlir::iree_compiler::IREE::GPU::symbolizeVirtualMMAIntrinsic(
+                enumValue);
+    if (virtualMmaIntrinsic) {
+      auto virtualMmaIntrinsicAttr =
+          mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsicAttr::get(
+              context, *virtualMmaIntrinsic);
+      mmaIntrinsicAttrs.push_back(virtualMmaIntrinsicAttr);
+      continue;
     }
-    targetInfo.mmaIntrinsics = wrap(builder.getArrayAttr(mmaIntrinsicAttrs));
+
+    assert(
+        false &&
+        ("Invalid MMA intrinsic value: " + std::to_string(enumValue)).c_str());
   }
+  targetInfo.mmaIntrinsics = wrap(builder.getArrayAttr(mmaIntrinsicAttrs));
 
   return targetInfo;
 }
+
+ireeGPUMMAIntrinsicResult
+ireeGPUTargetInfoGetMMAIntrinsics(MlirAttribute mmaIntrinsics) {
+  ireeGPUMMAIntrinsicResult result = {NULL, NULL, 0};
+  if (mlirAttributeIsNull(mmaIntrinsics) ||
+      !mlirAttributeIsAArray(mmaIntrinsics)) {
+    return result;
+  }
+
+  size_t numElements = mlirArrayAttrGetNumElements(mmaIntrinsics);
+  if (numElements == 0) {
+    return result;
+  }
+
+  result.mmaIntrinsicVals =
+      (mma_intrinsic_t *)malloc(numElements * sizeof(mma_intrinsic_t));
+  result.isVirtual = (bool *)malloc(numElements * sizeof(bool));
+  result.numMmaIntrinsics = numElements;
+
+  for (size_t i = 0; i < numElements; i++) {
+    MlirAttribute element = mlirArrayAttrGetElement(mmaIntrinsics, i);
+    if (ireeAttributeIsAGPUMMAIntrinsicAttr(element)) {
+      result.mmaIntrinsicVals[i] = ireeGPUMMAIntrinsicAttrGetValue(element);
+      result.isVirtual[i] = false;
+    } else if (ireeAttributeIsAGPUVirtualMMAIntrinsicAttr(element)) {
+      result.mmaIntrinsicVals[i] =
```

Comment:
Use early exits: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code